### PR TITLE
feat(button): added ripple button, issue #300 

### LINKED ISSUE #300: Button with hover animation
### Description
Create the component as shown in the video attached below.

### Animation Preview
- Screenshot/Video:

https://github.com/user-attachments/assets/aa36dcdf-7b55-4038-a714-135053a4d3c3

> [!NOTE]  
> Please refer to the **Animata contributing guidelines** available [here](https://www.animata.design/docs/contributing) for rules on how to contribute to the project. Let us know in the comments if you’re working on this issue, and feel free to ask any questions!

### Requirements
1. Create a new animated component that matches the provided design.
2. Add example usage of the component in the documentation or storybook.

<details>
  <summary>
    <h3> Guidelines & Best Practices</h3>
  </summary>

- Write reusable and clean code following the project’s style guide.
- The component should be **fully customizable**. The user should be able to modify props such as duration, easing, color, and size where applicable.
- Follow the [Contributing Guidelines](https://www.animata.design/docs/contributing) before submitting your pull request.

</details>

> [!IMPORTANT]  
> To ensure more contributors have the opportunity to participate, we kindly request that:
> - Each contributor submits only **one pull request** related to this issue.
> - If you're interested in working on this issue, please comment below. We will assign the issue on a **first-come, first-served basis**.

---

### Additional Resources
Posted on X by: https://x.com/codebydennis/status/1832041073897578894

---

> [!Tip]
> Join our Discord community if you need help or have any questions. We're here to help you get started and guide you through the process. [Join now](https://discord.gg/YfvqMf5MTE)

### PR DESCRIPTION
**Pull request by AE-Hertz**
I am attaching a working video- 

https://github.com/user-attachments/assets/2685d6b8-4273-4884-a93c-de0ff4b374f9

Please review it as its my first project working with storybook, i dont want to break your application by any means. 

I have also made this in react here is the github repo - https://github.com/AE-Hertz/Mouse-hover-effect 

Closes #300

### CODE REVIEW

**Comment by hari** [comment]
Please fix the TS error, you will need to cast the element.

```
./animata/button/ripple-button.tsx:40:14
Type error: Property 'style' does not exist on type 'Element'.

  38 |       const y = event.clientY - rect.top - size / 2;
  39 |
> 40 |       ripple.style.left = `${x}px`;
     |              ^
  41 |       ripple.style.top = `${y}px`;
  42 |       ripple.classList.remove("ripple-enter");
```

**Comment by AE-Hertz** [comment]
I fixed the casting issue, please check.

**Review by hari — CHANGES_REQUESTED** [review]
@AE-Hertz Please use tailwindcss.

**Comment by AE-Hertz** [comment]
@hari i am facing some issues and i am unable to figure out the reason - 
the working component need to have this much inline-css (working properly) - 
Screenshot 2024-10-08 231959
.
.
Here is the tailwind extend (not working properly) - 
Screenshot 2024-10-08 231931

Screenshot 2024-10-08 231950

issue facing (after removing keyframes and animation from inline-css) - 

https://github.com/user-attachments/assets/2917a485-2918-47a7-a12a-43fb1c02625a

If you can guide me on this , that would be great. Thank you

**Comment by AE-Hertz** [comment]
@sudhashrestha attaching a screenshot below, this much inlinecss is must to make this effect work, please let me know what can i do. 
Screenshot 2024-10-08 231959

**Comment by hari** [comment]
@AE-Hertz could you push the lastest changes so I can run it locally?

**Comment by AE-Hertz** [comment]
@hari yes i did, please check

**Comment by hari** [comment]
@AE-Hertz There is a bug when moving the mouse quickly. The ripple gets stuck. 

https://github.com/user-attachments/assets/959f6ea4-ec31-48ab-bae1-76a244054cd8

**Comment by AE-Hertz** [comment]
@hari i tried to resolve and after researching i found its actually the issue with react onMouseLeave() , read many article like this - https://stackoverflow.com/questions/29981236/how-do-you-hover-in-reactjs-onmouseleave-not-registered-during-fast-hover-ove its an old issue but i did not able to find any solution. 

If their is something with your knowledge please share with me , i will implement it.

**Comment by AE-Hertz** [comment]
@hari please check i think this method to  handle mouse event by checking the state is more consistent. 

If everything is fine, Please merge my PRs

**Comment by hari** [comment]
@AE-Hertz instead of state you can use ref to track the hovered state.

**Comment by AE-Hertz** [comment]
> @AE-Hertz instead of state you can use ref to track the hovered state.

yeah , actually. should i change it then?

**Comment by hari** [comment]
@AE-Hertz sure, why not?

**Comment by AE-Hertz** [comment]
@hari hey i implemented it with useRef but its not as consistent as this version(useState), thats why not pushing that code(useRef) . 

see clip. 

https://github.com/user-attachments/assets/15f757e4-091f-4eb9-a64d-9f96e425fe9b

**Comment by AE-Hertz** [comment]
@hari well, i used both useRef and useState for better consistency, sending PR .

**Review by sudhashrestha — APPROVED** [review]
Looks good @AE-Hertz

File: animata/button/ripple-button.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        }
+        @keyframes ripple-enter {
+          from { transform: scale(0); opacity: 1; }
+          to { transform: scale(1); opacity: 1; }
+        }
+        @keyframes ripple-leave {
+          from { transform: scale(1); opacity: 1; }
+          to { transform: scale(0); opacity: 1; }
+        }
+      `}</style>
```
Comment (sudhashrestha) [inline animata/button/ripple-button.tsx:124]: @AE-Hertz Just noticed this. Please use tailwind CSS instead of this.

**Review by sudhashrestha — CHANGES_REQUESTED** [review]

File: animata/button/ripple-button.tsx (hunk 1 above)
Comment (AE-Hertz) [reply animata/button/ripple-button.tsx:124]: @sudhashrestha we were talking about the same thing with hari , as this much inlinecss is necessary to work this effect, i tried everything and extending twconfig is also not working, can you guide me on this as i tried everything but results are same that this much inline is needed.

**Review by hari — APPROVED** [review]

**Review by hari — APPROVED** [review]

**Review by sudhashrestha — APPROVED** [review]

**Pull request merged**